read_zdf_vis_3d.py

```diff
@@ -42,14 +36,64 @@ def _main():
     )
     plt.colorbar()
     plt.title("Depth map")
-    plt.show()
+    plt.show(block=False)
+
+
+def _get_mid_point(xyz):
+    """Calculate mid point from average of the 100 centermost points.
+
+    Args:
+        xyz: X,Y and Z co-ordinates
+
+    Returns:
+        mid_point: Calculated
+
+    """
+    xyz_center = xyz[
+        int(xyz.shape[0] / 2 - 5) : int(xyz.shape[0] / 2 + 5),
+        int(xyz.shape[1] / 2 - 5) : int(xyz.shape[1] / 2 + 5),
+        :,
+    ]
+    xyz_center[xyz_center == 0] = np.nan
+    return (
+        np.nanmedian(xyz_center[:, :, 0]),
+        np.nanmedian(xyz_center[:, :, 1]),
+        np.nanmedian(xyz_center[:, :, 2]),
+    )
+
+
+def _display_pointcloud(rgb, xyz):
```

Comment:
Should we add few comments for this function? Lines 69 and 72 in particular for me were not straightforward to understand what is happening.

Reply:
Yes, absolutely